### PR TITLE
Re-adds escape pods

### LINKED ISSUE #1741: Sabre and Sabre Associated Canopy Bugs
## Testmerges:
#1726 
#1714 
#1692 
#1737
#1730 

## Reproduction:

I haven't done this myself, but judging from what people who have suffered this bug, I believe this is how it occurs. 

1: Enter a sabre
2: Buckle in at the helm
3: Use the helm console
4: Attempt to get off the helm console
5: Bump your head repeatedly on the canopy. (Apparently they can't open it.)
6: Get stuck forever.

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
Makes fighters blow up into escape pods again. I just made it so they can always move and can't get deleted with people inside.
Fixes #1741

## Why It's Good For The Game
This makes more sense than teleporting people back to the mothership and it's on the roadmap

## Changelog
:cl:
add: Fighter escape pods are back
fix: They don't get shot down anymore 
fix: Removes engine trouble
/:cl:

### CODE REVIEW

File: nsv13/code/game/general_quarters/dropship.dm [hunk 1]
```diff
@@ -151,53 +151,6 @@
 		if(M && M.client && M.hud_used && length(M.client.parallax_layers))
 			M.hud_used.update_parallax(force=TRUE)
 
-//Jank ass override, because this is actually necessary... but eughhhh
```
Comment (covertcorvid) [inline nsv13/code/game/general_quarters/dropship.dm:154]: All of this was also getting done elsewhere? Seemed to work when I was testing stuff but obviously I can't try it with multiple occupants very well

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 2]
```diff
@@ -490,24 +515,73 @@ Been a mess since 2018, we'll fix it someday (probably)
...
+			last_pilot.forceMove(escape_pod)
+			escape_pod.start_piloting(last_pilot, "pilot")
+			escape_pod.attack_hand(last_pilot) // Bring up UI
+			mobs_in_ship -= last_pilot
+			escape_pod.mobs_in_ship += last_pilot
+			last_pilot.overmap_ship = escape_pod
+
+		for(var/mob/M in mobs_in_ship)
+			M.forceMove(escape_pod)
+			if(!escape_pod.pilot || escape_pod.pilot.stat == DEAD || escape_pod.pilot.incapacitated()) // Someone please drive this thing
```
Comment (TheFakeElon) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:574]: `incapacitated()` checks for stat :)
```suggestion
			if(!escape_pod.pilot || escape_pod.pilot.incapacitated()) // Someone please drive this thing
```

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 3]
```diff
@@ -490,24 +515,73 @@ Been a mess since 2018, we'll fix it someday (probably)
...
+		return
+	// Create pod
+	var/obj/structure/overmap/fighter/escapepod/escape_pod = new path(get_turf(src))
+	if(!istype(escape_pod))
+		message_admins("Unable to create escape pod for [src] with path [path]")
+		return
+	escape_pod.name = "[name] - escape pod"
+	escape_pod.faction = faction
+	escape_pod.desired_angle = 0
+	escape_pod.user_thrust_dir = NORTH
```
Comment (TheFakeElon) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:556]: would be cool if it inherited momentum from it's fighter 😎

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 4]
```diff
@@ -490,24 +515,73 @@ Been a mess since 2018, we'll fix it someday (probably)
...
+		qdel(src)
+
+/obj/structure/overmap/fighter/proc/create_escape_pod(path, mob/last_pilot)
+	if(!ispath(path))
+		return
+	// Create pod
+	var/obj/structure/overmap/fighter/escapepod/escape_pod = new path(get_turf(src))
+	if(!istype(escape_pod))
+		message_admins("Unable to create escape pod for [src] with path [path]")
+		return
```
Comment (TheFakeElon) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:552]:
```suggestion
		qdel(escape_pod)
		return
```

File: nsv13/code/modules/overmap/fighters/_fighters.dm (hunk 3 above)
Comment (TheFakeElon) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:556]: would be cool if it inherited momentum from it's fighter 😎

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 5]
```diff
@@ -490,24 +515,73 @@ Been a mess since 2018, we'll fix it someday (probably)
...
 	return TRUE
 
+/obj/structure/overmap/fighter/escapepod/eject(mob/living/M, force=FALSE)
+	. = ..()
+	if(. && !length(mobs_in_ship)) // Last one out means we don't need this anymore
+		qdel(src)
+
+/obj/structure/overmap/fighter/proc/create_escape_pod(path, mob/last_pilot)
+	if(!ispath(path))
+		return
```
[the remark this answers is not in the document]
Comment (covertcorvid) [reply nsv13/code/modules/overmap/fighters/_fighters.dm:547]: I wanted admins to be able to use the proc if they ever needed to

File: nsv13/code/modules/overmap/fighters/_fighters.dm (hunk 3 above)
Comment (covertcorvid) [reply nsv13/code/modules/overmap/fighters/_fighters.dm:556]: they're supposed to be really slow though 🤔

File: nsv13/code/modules/overmap/fighters/_fighters.dm (hunk 5 above)
[the remark this answers is not in the document]
Comment (TheFakeElon) [reply nsv13/code/modules/overmap/fighters/_fighters.dm:544]: Admins creating runtimes with debug isn't a big issue as they can do it with basically any proc in the game. The check is pretty pointlless

**Review by Karmic-Skink — APPROVED** [review]
As I currently am unaware of any issues with this currently, I will approve.

Additional feedback would be fantastic.

**Pull request merged**